### PR TITLE
IGNITE-18220: Implement integration tests for completers

### PR DESCRIPTION
**Pull request by PakhomovAlexander**
https://issues.apache.org/jira/browse/IGNITE-18220

### CODE REVIEW

File: modules/cli/src/integrationTest/java/org/apache/ignite/internal/cli/commands/CliCommandTestNotInitializedIntegrationBase.java [hunk 1]
```diff
@@ -106,6 +107,10 @@ protected void execute(String... args) {
         exitCode = cmd.execute(args);
     }
 
+    protected CommandLine getCmd() {
```
Comment (Flaugh24) [inline modules/cli/src/integrationTest/java/org/apache/ignite/internal/cli/commands/CliCommandTestNotInitializedIntegrationBase.java:110]: Why do you need this method? You can just make cmd protected.

Comment (PakhomovAlexander) [reply modules/cli/src/integrationTest/java/org/apache/ignite/internal/cli/commands/CliCommandTestNotInitializedIntegrationBase.java:110]: I wouldn't like to give an access to fields event for children. But I could rename the method to `cmd()`

File: modules/cli/src/integrationTest/java/org/apache/ignite/internal/cli/commands/CliCommandTestInitializedIntegrationBase.java [hunk 2]
```diff
@@ -31,10 +32,14 @@ public class CliCommandTestInitializedIntegrationBase extends CliCommandTestNotI
...
+        initializeCluster(metaStorageNodeName(testInfo));
     }
 
     protected String metaStorageNodeName(TestInfo testInfo) {
         return testNodeName(testInfo, 0);
     }
+
+    protected List<String> allNodeNames() {
+        return CLUSTER_NODE_NAMES;
+    }
```
Comment (Pochatkin) [inline modules/cli/src/integrationTest/java/org/apache/ignite/internal/cli/commands/CliCommandTestInitializedIntegrationBase.java:44]: Why is it here? This static field in IgniteTestBase has protected modifier and you expand it by NON static protected method in some child class, absolutely don't understand what the reason

File: modules/cli/src/integrationTest/java/org/apache/ignite/internal/cli/commands/CliCommandTestNotInitializedIntegrationBase.java [hunk 3]
```diff
@@ -77,6 +77,7 @@ public class CliCommandTestNotInitializedIntegrationBase extends IntegrationTest
     @BeforeEach
     public void setUp(TestInfo testInfo) throws Exception {
         super.setUp(testInfo);
+
```
Comment (Pochatkin) [inline modules/cli/src/integrationTest/java/org/apache/ignite/internal/cli/commands/CliCommandTestNotInitializedIntegrationBase.java:80]: Why empty line?

File: modules/cli/src/main/java/org/apache/ignite/internal/cli/core/repl/completer/NodeUrlProvider.java [hunk 4]
```diff
@@ -40,7 +40,7 @@ public NodeUrlProvider(Session session, ConfigManagerProvider configManagerProvi
     }
 
     /** Resolves the url for given words. */
-    public String resolveUrl(String[] words) {
+    public String resolveUrl(String... words) {
```
Comment (Pochatkin) [inline modules/cli/src/main/java/org/apache/ignite/internal/cli/core/repl/completer/NodeUrlProvider.java:43]: What the point of vararg here? So, lets imagine that its required change but why you didn't change usages of this method. For example org/apache/ignite/internal/cli/core/repl/completer/hocon/ClusterConfigDynamicCompleterFactory.java:58 still use String array initialization

File: modules/cli/src/integrationTest/java/org/apache/ignite/internal/cli/commands/CliCommandTestInitializedIntegrationBase.java (hunk 2 above)
Comment (PakhomovAlexander) [reply modules/cli/src/integrationTest/java/org/apache/ignite/internal/cli/commands/CliCommandTestInitializedIntegrationBase.java:44]: Agreed to add the `static` modifier to the method signature.

**Comment by Pochatkin** [comment]
LGTM

**Pull request closed without merging**